zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/DeploymentReconstructProcessor.java

```diff
@@ -95,6 +108,9 @@ public void processRecord(final TypedRecord<DeploymentRecord> record) {
     }
     stateWriter.appendFollowUpEvent(
         deploymentRecord.getDeploymentKey(), DeploymentIntent.RECONSTRUCTED, deploymentRecord);
+    // trigger reconstruction of another deployment reconstruction
+    // TODO: is there a "safe" way to avoid allocating a new DeploymentRecord each time?
```

Comment:
`appendNewCommand` eventually copies the record(see `RecordBatchEntry#createEntry`) so it should be safe to use one shared instance per partition. It just can't be static because then we'd have concurrent access on the same object from multiple partitions.

Reply:
I'll create a local instance then :+1: